src/Elasticsearch/Endpoints/Indices/Mapping/Put.php

```diff
@@ -41,19 +41,21 @@ public function setBody($body)
      */
     public function getURI()
     {
-        if (isset($this->type) !== true) {
+        $index = $this->index ?? null;
+        $type = $this->type ?? null;
+
+        if (null === $index && $type === $index) {
             throw new Exceptions\RuntimeException(
-                'type is required for Put'
+                'type or index are required for Put'
```

Comment:
shouldn't it be `'type and index are required for Put'`?

Reply:
Actually, in ES 6.7 the {type} part is not required anymore. If you see the available paths [here](https://github.com/elastic/elasticsearch/blob/v6.7.0/rest-api-spec/src/main/resources/rest-api-spec/api/indices.put_mapping.json#L6-L7) is it possible to request `{index}/_mapping`, `/_mapping/{type}`, `{index}/_mappings/{type}`, etc.